Spigot-Server-Patches/0551-Add-BlockDestroyedByNeighborEvent.patch

```diff
@@ -0,0 +1,193 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tom <cryptite@gmail.com>
+Date: Sun, 9 Aug 2020 09:12:26 -0500
+Subject: [PATCH] Add BlockDestroyedByNeighborEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/BlockBase.java b/src/main/java/net/minecraft/server/BlockBase.java
+index 7b0cc7833448a9e77e85be10f4e02ce43fdead58..d91cf7a951c793211d2ee170cd0f5476c8e34cc3 100644
+--- a/src/main/java/net/minecraft/server/BlockBase.java
++++ b/src/main/java/net/minecraft/server/BlockBase.java
+@@ -525,6 +525,19 @@ public abstract class BlockBase {
+ 
+                 blockposition_mutableblockposition.a((BaseBlockPosition) blockposition, enumdirection);
+                 IBlockData iblockdata = generatoraccess.getType(blockposition_mutableblockposition);
++
++                // Paper start - Propagate the pendingPlayerBlockEvents
++                World world = generatoraccess.getMinecraftWorld();
++                if (getMaterial() == Material.AIR && !iblockdata.canPlace(generatoraccess, blockposition_mutableblockposition)) {
```

Comment:
`Material#isAir` is going to be *vital* to be ready for the 1.17 update.

Reply:
nms Material doesn't have an isAir check. Not sure what should be done here.